source/common/upstream/upstream_impl.cc

```diff
@@ -846,9 +846,14 @@ void MainPrioritySetImpl::updateCrossPriorityHostMap(const HostVector& hosts_add
   }
 }
 
-LazyClusterTrafficStats ClusterInfoImpl::generateStats(Stats::Scope& scope,
-                                                       const ClusterTrafficStatNames& stat_names) {
-  return std::make_unique<ClusterTrafficStats>(stat_names, scope);
+std::unique_ptr<LazyCompatibleClusterTrafficStats>
+ClusterInfoImpl::generateStats(Stats::ScopeSharedPtr scope,
+                               const ClusterTrafficStatNames& stat_names, bool lazyinit) {
+  if (lazyinit) {
+    return std::make_unique<LazyInitClusterTrafficStats>(stat_names, scope);
```

Comment:
WDYT of making a factory method or helper function to do the above?

You went through the trouble of creating this re-usable LazyCompatibleInterface class hierarchy - you might as well standardize how it gets constructed.

I'd do this as a static method in LazyCompatibleInterface.

Reply:
I think you are suggesting a template method in LazyCompatibleInterface. 

I think it's probably fine since this is gonna be a one line init for any future lazyStats. An factory method is just to make reader to jump to the source file to figure what happens there.